src/travertino/node.py

```diff
@@ -14,6 +18,71 @@ def __init__(self, style, applicator=None, children=None):
             for child in children:
                 self.add(child)
 
+    @property
+    def style(self):
+        """The node's style.
+
+        Assigning a style triggers an application of that style if an applicator has
+        already been assigned.
+        """
+
+        return getattr(self, "_style", None)
+
+    @style.setter
+    def style(self, style):
+        if style is None:
+            self._style = None
+            return
+
+        self._style = style.copy()
+        self.intrinsic = self.style.IntrinsicSize()
+        self.layout = self.style.Box(self)
+
+        if self.applicator:
+            self.style._applicator = self.applicator
+
+            try:
+                self.style.reapply()
+            # Backwards compatibility for Toga
+            except AttributeError:
+                warn(
+                    "Failed to apply style, when new style assigned and applicator "
+                    "already present. Node should be sufficiently initialized to "
+                    "apply style before it has both a style and an applicator.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
+
+    @property
+    def applicator(self):
+        """This node's applicator, which handles applying the style.
+
+        Assigning an applicator triggers an application of that style if a style has
+        already been assigned.
+        """
+        return getattr(self, "_applicator", None)
```

Comment:
As with style  - I think we can just use `self._applicator` here.

Reply:
I thought you were right, until I tried it and realized that it is accessed in the setter for style. (It's nothing special about applicator vs. style, except for the fact that this happens to be the one the initializer assigns to last.)

Reply:
In which case, I'd be inclined to prime the property in the constructor. Explicitly setting `self._applicator` as an initial step means we're explicitly acknowledging the circular dependency, rather than implicitly relying on a fallback.